torch/nn/modules/conv.py

```diff
@@ -530,8 +532,9 @@ def __init__(self, in_channels, out_channels, kernel_size, stride,
 
     # dilation being an optional parameter is for backwards
     # compatibility
-    def _output_padding(self, input, output_size, stride, padding, kernel_size, dilation=None):
-        # type: (Tensor, Optional[List[int]], List[int], List[int], List[int], Optional[List[int]]) -> List[int]
+    def _output_padding(self, input: Tensor, output_size: Optional[List[int]],
+                        stride: List[int], padding: List[int], kernel_size: List[int],
+                        dilation: Optional[List[int]] = None) -> List[int]:
```

Comment:
```suggestion
    def _output_padding(self, input: Tensor, output_size: Optional[List[int]],
                        stride: Tuple[int, ...], padding: Tuple[int, ...], kernel_size: Tuple[int, ...],
                        dilation: Optional[Tuple[int, ...]] = None) -> Tuple[int, ...]:
```

Then change the final return to 
```
-        return ret
+        return tuple(ret)
```

Reply:
Thanks! I've updated the source code with the suggested changes

Reply:
I'll revert the commit since one of the jit tests is failing with the message:
```
RuntimeError: 
Expression of type dots cannot be used in a type expression:
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 536
    def _output_padding(self, input: Tensor, output_size: Optional[List[int]],
                        stride: Tuple[int, ...], padding: Tuple[int, ...], kernel_size: Tuple[int, ...],
                                           ~~~ <--- HERE
                        dilation: Optional[Tuple[int, ...]] = None) -> Tuple[int, ...]:
        if output_size is None:
'ConvTranspose1d._output_padding' is being compiled since it was called from 'ConvTranspose1d.forward'
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 688
            raise ValueError('Only `zeros` padding mode is supported for ConvTranspose1d')
    
        output_padding = self._output_padding(
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            input, output_size, self.stride, self.padding, self.kernel_size, self.dilation)
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        return F.conv_transpose1d(
            input, self.weight, self.bias, self.stride, self.padding,
```

Reply:
> `Expression of type dots cannot be used in a type expression:`

That seems to disagree with the `Tuple` docs in https://pytorch.org/docs/master/jit_language_reference.html#supported-type. Looks like only fixed-length tuples are supported? If so, would be good to fix that docs entry.

This would explain my question above about everything being typed `List[int]`.

Reply:
yes. i agree the doc is confusing.. it should be `Tuple[T1, T2]` instead of `Tuple[T1, T2, ...]` which means a variable-length tuple, see https://docs.python.org/3/library/typing.html#typing.Tuple

Reply:
did a bit of research. seems like in fact it is only afffecting `Tuple[T, ...]` annotation. i checked that the following actually worked so JIT can parse ellipsis correct.
```
def foo():
            n = torch.ones(16).reshape(2,2,2,2)
            return n[1,...,1]  # equivalent to return n[1,:,:,1]
```